pkg/apis/cilium/types_network.go

```diff
@@ -138,4 +138,6 @@ type NetworkConfig struct {
 	IPv6 *IPv6
 	// BPFSocketLBHostnsOnly flag to be enabled or not
 	BPFSocketLBHostnsOnly *BPFSocketLBHostnsOnly
+	// EgressGateway flag to be enabled or not
+	EgressGateway *bool
```

Comment:
```suggestion
	// EgressGateway flag to be enabled or not
	EgressGateway *EgressGateway
```

// EgressGateway enablement for cilium
type EgressGateway struct {
	Enabled bool
}